temporal/server_option.go

```diff
@@ -77,6 +77,13 @@ func ForServices(names []string) ServerOption {
 	})
 }
 
+// WithStaticHosts disables dynamic service membership and resolves service addresses statically from the config
+func WithStaticHosts(hosts map[primitives.ServiceName]string) ServerOption {
```

Comment:
If we're offering a server option, I think we should offer the more flexible version that supports multiple hosts. Maybe something like
```suggestion
type ServiceHosts struct {
    All  []string // Addresses of all hosts for this service. Must contain at least one entry for all required services (frontend, history, matching, worker).
    Self string   // Address of this host. Must be present for the services passed to ForServices.
}
func WithStaticMembership(hosts map[primitives.ServiceName]ServiceHosts) ServerOption {
```
or with two parallel maps if you want, since the Self map is allowed to be smaller.